src/librustdoc/html/render/write_shared.rs

```diff
@@ -356,15 +357,14 @@ if (typeof exports !== 'undefined') {exports.searchIndex = searchIndex};
 
             let content = format!(
                 "<h1>List of all crates</h1><ul class=\"all-items\">{}</ul>",
-                krates
-                    .iter()
-                    .map(|s| {
-                        format!(
-                            "<li><a href=\"{trailing_slash}index.html\">{s}</a></li>",
-                            trailing_slash = ensure_trailing_slash(s),
-                        )
-                    })
-                    .collect::<String>()
+                krates.iter().fold(String::new(), |mut output, s| {
+                    _ = write!(
+                        output,
+                        "<li><a href=\"{trailing_slash}index.html\">{s}</a></li>",
+                        trailing_slash = ensure_trailing_slash(s),
+                    );
+                    output
+                })
```

Comment:
[`Itertools::format_with`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.format_with) could have this write directly into the `{}` above.